src/linear_curve.rs

```diff
@@ -0,0 +1,74 @@
+use cgmath::{Vector2, Vector3, Vector4};
+
+use buffer::Buffer;
+use device::Device;
+use geometry::Geometry;
+use sys::*;
+use {BufferType, Format, GeometryType};
+
+pub struct LinearCurve<'a> {
+    device: &'a Device,
+    pub(crate) handle: RTCGeometry,
+    pub vertex_buffer: Buffer<'a, Vector4<f32>>,
+    pub index_buffer: Buffer<'a, u32>,
+    pub flag_buffer: Buffer<'a, u32>,
+}
+
+impl<'a> LinearCurve<'a> {
+    pub fn unanimated(device: &'a Device, num_segments: usize, num_verts: usize, curve_type: usize) -> LinearCurve<'a> {
+        let h: RTCGeometry;
+        match curve_type {
+        _ => h = unsafe { rtcNewGeometry(device.handle, GeometryType::FLAT_LINEAR_CURVE) },
+        };
+        let mut vertex_buffer = Buffer::new(device, num_verts);
+        let mut index_buffer = Buffer::new(device, num_segments);
+        let mut flag_buffer = Buffer::new(device, num_verts);
+        unsafe {
+            rtcSetGeometryBuffer(
+                h,
+                BufferType::VERTEX,
+                0,
+                Format::FLOAT4,
+                vertex_buffer.handle,
+                0,
+                16,
+                num_verts,
+            );
+            vertex_buffer.set_attachment(h, BufferType::VERTEX, 0);
+
+            rtcSetGeometryBuffer(
+                h,
+                BufferType::INDEX,
+                0,
+                Format::UINT,
+                index_buffer.handle,
+                0,
+                4,
+                num_segments,
+            );
+            index_buffer.set_attachment(h, BufferType::INDEX, 0);
+
+            rtcSetGeometryBuffer(
+                h,
+                BufferType::FLAGS,
+                0,
+                Format::UCHAR,
+                flag_buffer.handle,
+                0,
+                4,
```

Comment:
Flags stride should be 1 since it's uchars. The documentation for [`rtcSetGeometryBuffer`](https://www.embree.org/api.html#rtcsetgeometrybuffer) mentions that the byteStride must be aligned to 4, but this doesn't apply to the special "flags" buffer which should be bytes with a stride of 1. In their example they also use a stride of 1 https://github.com/embree/embree/blob/master/tutorials/curve_geometry/curve_geometry_device.cpp#L96 . Here Embree's `CurveGeometry` they only apply the alignment restriction to non-flags buffers: https://github.com/embree/embree/blob/master/kernels/common/scene_curves.cpp#L59-L60